app/main.py

```diff
@@ -16,4 +16,17 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def create_sorted_variables(*args) -> dict:
+    sorted_variables = {"mutable": [], "immutable": []}
+
+    for arg in args:
+        if isinstance(arg, (float, int, str, bool, tuple)):
+            sorted_variables["immutable"].append(arg)
+        else:
+            sorted_variables["mutable"].append(arg)
```

Comment:
The logic in your `isinstance` check is not entirely correct. You're verifying if a variable is of type `float`, `int`, `str`, `bool`, or `tuple`, and if so, you're treating it as immutable. However, this logic would fail for instances of custom classes or other complex data types. Also, you're treating all other types as mutable, but what about types like `frozenset` which are also immutable? Here's a suggestion: instead of checking for immutability, check for mutability. Python has fewer mutable built-in types: `list`, `set`, `dict`, and most custom classes. So, you can check if a variable is an instance of those types, and if so, treat it as mutable. Otherwise, treat it as immutable.